bluesky/plans.py

```diff
@@ -133,55 +172,66 @@ def list_scan(detectors, *args, per_step=None, md=None):
-                       list(lengths.values()))
+    length_check = all(
+        elem == list(lengths.values())[0] for elem in list(lengths.values())
+    )
 
     if not length_check:
-        raise ValueError("The lengths of all lists in *args must be the same. "
-                         "However the lengths in args are : "
-                         "{}".format(lengths))
-
-    md_args = list(chain(*((repr(motor), pos_list)
-                           for motor, pos_list in partition(2, args))))
+        raise ValueError(
+            "The lengths of all lists in *args must be the same. "
+            "However the lengths in args are : "
+            "{}".format(lengths)
+        )
+
+    md_args = list(
+        chain(*((repr(motor), pos_list) for motor, pos_list in partition(2, args)))
+    )
     motor_names = list(lengths.keys())
 
-    _md = {'detectors': [det.name for det in detectors],
-           'motors': motor_names,
-           'num_points': length,
-           'num_intervals': length - 1,
-           'plan_args': {'detectors': list(map(repr, detectors)),
-                         'args': md_args,
-                         'per_step': repr(per_step)},
-           'plan_name': 'list_scan',
-           'plan_pattern': 'inner_list_product',
-           'plan_pattern_module': plan_patterns.__name__,
-           'plan_pattern_args': dict(args=md_args),
-           'hints': {},
-           }
+    _md = {
+        "detectors": [det.name for det in detectors],
+        "motors": motor_names,
+        "num_points": length,
+        "num_intervals": length - 1,
+        "plan_args": {
+            "detectors": list(map(repr, detectors)),
+            "args": md_args,
+            "per_step": repr(per_step),
+        },
+        "plan_name": "list_scan",
+        "plan_pattern": "inner_list_product",
+        "plan_pattern_module": plan_patterns.__name__,
+        "plan_pattern_args": dict(args=md_args),
+        "hints": {},
+    }
     _md.update(md or {})
 
     x_fields = []
     for motor in motors:
         x_fields.extend(get_hinted_fields(motor))
 
-    default_dimensions = [(x_fields, 'primary')]
+    default_dimensions = [(x_fields, "primary")]
 
     default_hints = {}
     if len(x_fields) > 0:
         default_hints.update(dimensions=default_dimensions)
 
     # now add default_hints and override any hints from the original md (if
     # exists)
-    _md['hints'] = default_hints
-    _md['hints'].update(md.get('hints', {}) or {})
+    _md["hints"] = default_hints
+    _md["hints"].update(md.get("hints", {}) or {})
 
     full_cycler = plan_patterns.inner_list_product(args)
 
-    return (yield from scan_nd(detectors, full_cycler, per_step=per_step,
-                               md=_md))
+    return (yield from scan_nd(detectors, full_cycler, per_step=per_step, md=_md))
 
 
-def rel_list_scan(detectors, *args, per_step=None, md=None):
+def rel_list_scan(
+    detectors: List[Readable],
+    *args: Any,
```

Comment:
Here is another issue with the Queue Server, since it will not attempt to convert strings to objects (motors in this particular example) if parameter is annotated with a type not associated with Ophyd devices (such as `Readable`, `Movable` etc). So in this case `*args` should have no annotation (by default the queue server tries to convert all strings to objects) or the type should be something like `Union[Movable, Any]`. It should also work with mypy.

Reply:
Interesting, for blueapi we went the opposite way. Only attempts to convert strings if they are annotated with protocol types or things that follow protocol types.

Reply:
This is what the Queue Server does. But it also attempts to convert all strings it finds in values of parameters that are not annotated. This may lead to errors, since any string matching a plan or a device name will be converted to object, but this never happened in practice. Unfortunately, our local users do not want to learn about type annotations, which makes this feature very useful.

Reply:
So the parameter annotated with `typing.Any` is considered having type and the Queue Server does not try to convert strings to objects.